src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,150 @@
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineMethod;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InlineMethod
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.InlineMethod)), Shared]
+    [Export(typeof(CSharpInlineMethodRefactoringProvider))]
+    internal sealed class CSharpInlineMethodRefactoringProvider :
+        AbstractInlineMethodRefactoringProvider<MethodDeclarationSyntax, StatementSyntax, ExpressionSyntax, InvocationExpressionSyntax>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpInlineMethodRefactoringProvider()
+            : base(CSharpSyntaxFacts.Instance, CSharpSemanticFactsService.Instance)
+        {
+        }
+
+        protected override ExpressionSyntax? GetInlineExpression(MethodDeclarationSyntax methodDeclarationSyntax)
+        {
+            var blockSyntaxNode = methodDeclarationSyntax.Body;
+            if (blockSyntaxNode != null)
+            {
+                // 1. If it is an ordinary method with block
+                var blockStatements = blockSyntaxNode.Statements;
+                if (blockStatements.Count == 1)
+                {
+                    var statementSyntax = blockStatements[0];
+                    return statementSyntax switch
+                    {
+                        // Note: For this case this will return null in Callee()
+                        // void Caller() { Callee(); }
+                        // void Callee() { return; }
+                        // Refactoring won't be provided for this case.
+                        ReturnStatementSyntax returnStatementSyntax => returnStatementSyntax.Expression,
+                        ExpressionStatementSyntax expressionStatementSyntax => expressionStatementSyntax.Expression,
+                        ThrowStatementSyntax throwStatementSyntax => throwStatementSyntax.Expression,
+                        _ => null
+                    };
+                }
+            }
+            else
+            {
+                // 2. If it is an Arrow Expression
+                var arrowExpressionNode = methodDeclarationSyntax.ExpressionBody;
+                return arrowExpressionNode?.Expression;
+            }
+
+            return null;
+        }
+
+        protected override SyntaxNode? GetEnclosingMethodLikeNode(SyntaxNode syntaxNode)
+        {
+            for (var node = syntaxNode; node != null; node = node.Parent)
+            {
+                if (node is BaseMethodDeclarationSyntax
+                    || node.IsKind(SyntaxKind.LocalFunctionStatement)
+                    || node is LambdaExpressionSyntax)
```

Comment:
is this to find the container of the caller, is the method-decl for the callee?  i woudl add Caller/Callee to the name to help make this clearer.

if this is for the caller, do you also handle single/multi-line lambdas in teh VB case?

Reply:
probably want to check for AnonymousFunctionExpressionSyntax not LambdaExpressionSyntax.

Reply:
This is only used for Callee